packages/forma-36-react-components/src/components/Accordion/AccordionHeader/AccordionHeader.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { FC } from 'react';
+import cn from 'classnames';
+
+import Subheading from '../../Typography/Subheading';
+import Icon from '../../Icon';
+
+import styles from '../Accordion.css';
+
+interface AccordionHeaderProps {
+  /**
+   * Child nodes to be rendered in the component
+   */
+  children?: React.ReactNode;
+  /**
+   * An ID used for testing purposes applied as a data attribute (data-test-id)
+   */
+  testId?: string;
+  /**
+   * The function that will be called once the user clicks on the accordion title
+   */
+  handleClick: VoidFunction;
+  /**
+   * A boolean that tells if the accordion should be expanded or collapsed
+   */
+  isExpanded: boolean;
+  /**
+   * An unique id that is necessary for the aria roles and properties
+   */
+  ariaId: number | null;
+}
+
+export const AccordionHeader: FC<AccordionHeaderProps> = ({
+  children,
+  handleClick,
+  isExpanded,
+  ariaId,
+}: AccordionHeaderProps) => {
+  const classNames = cn(styles.AccordionHeader, {
+    [styles['AccordionHeader--expanded']]: isExpanded,
+  });
+
+  return (
+    <Subheading>
```

Comment:
Should be able to pass an `element` prop to the Subheading component as by default it'll use a `h2`, can imagine wanting control over this. Alternatively, could use `details`/`summary` elements https://caniuse.com/#feat=details

Reply:
@mohamedturki also suggested using details/summary. I will take a look on that

Reply:
I discovered something about using details/summary and now I don't know what is the best option

When testing with Mac's VoiceOver tool I couldn't navigate using headings because the heading now needed to be inside the summary (the first child of details needs to be summary) and clicking on the heading won't expand the accordion panel since the clickable element is the summary, not the heading

I got around this by using the same aria properties I used for the button solution. You can navigate to the heading correctly, but the VoiceOver does not let you go to the panel to read its content and I think the reason is something to do with [the way summary deals with headings](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/summary#Summaries_as_headings)

I'm committing the version with details/summary so you all can test and maybe find a solution

In case we can't solve the heading navigation I would suggest keeping the button/heading solution because it offers a better a11y experience

(in any case, we create the `element` prop)

Reply:
Also, I'm getting this on Travis but not locally

![Screenshot 2020-08-04 at 16 47 43](https://user-images.githubusercontent.com/6597467/89308153-3ca0ac00-d672-11ea-892e-b2c19d837c4f.png)

`jsx-a11y/click-events-have-key-events`: summary tag works with keyboard by default, I don't understand why we would need to add key events
`jsx-a11y/interactive-supports-focus`: summary and details are tabbable but they are not in the [rules list](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/master/docs/rules/interactive-supports-focus.md) apparently 

☝️  I would add up these to the cons of using details/summary for now

Reply:
@gui-santos I guess you meant @mshaaban0 :)

Reply:
😆  yes I did, @mohamedturki! I wrote "@moe" and selected the first name but that's you... sorry

@mshaaban0 I swear I know your name 🙏

Reply:
@gui-santos although you just mentioned a third person "moe" 😅 but thank you for implementing this.

Regarding, the details/summary, I'm wondering what is the use case for headings inside the summary tag? headings are meant to be identifiers for sections, and big blocks, I think the behaviour is normal as headings inside buttons isn't valid html5 in the first place. 

Regarding the a11y errors, maybe removing the custom aria roles on summary would fix it?

Reply:
> @gui-santos although you just mentioned a third person "moe" 😅  

🤦 

The case for the headings was more trying to follow [these directives from WAI-ARIA](https://www.w3.org/TR/wai-aria-practices-1.2/#wai-aria-roles-states-and-properties) which assume we are using buttons and headings. 
But if heading navigation is not something that the accordion should worry about then that's solved I guess hahaha

Uhmm I will remove the aria roles later, but I think I had a different problem when I didn't have them. I will need to test again to remember, but I think I was having a problem going to the accordion content using the VoiceOver. Let me check again and I post it here

Reply:
Forget everything I just said! It worked great with details and summary

Thanks you all!